test/e2e/framework/util.go

```diff
@@ -187,6 +188,13 @@ const (
 	// images within this time we simply log their output and carry on
 	// with the tests.
 	ImagePrePullingTimeout = 5 * time.Minute
+
+	// CosOSImage is the string advertised by the OSImage field of a node running COS
+	CosOSImage = "Container-Optimized OS from Google"
+
+	// GPUResourceName is the extended name of the GPU resource since v1.8
+	// this uses the device plugin mechanism
+	NVIDIAGPUResourceName = "nvidia.com/gpu"
```

Comment:
Let's not add this here. We don't want to dictate what nvidia chooses as its resource name for device plugins.

Reply:
I agree  that we don't want to dictate what nvidia chooses as its resource name. 
However, right now this the resource name advertised by the device plugin @jiayingz implemented.

Do you have another suggestion :) ?

Reply:
I would move these methods out from framework/util.go and put them back where they are getting used. As far as I understand testing based on a vendor specific device plugin is temporary. Eventually we will use a dummy device plugin to test that the device plugin functionality in kubelet works as expected. And each vendor would have their own tests for device plugins.

Reply:
> As far as I understand testing based on a vendor specific device plugin is temporary

Right. However I don't want to duplicate these methods as this is now part of e2e_node and they were used in e2e/scheduling.

> Eventually we will use a dummy device plugin to test that the device plugin functionality in kubelet works as expected

As mentioned by @jiayingz I think we still want these to be used in a real scenario such as NVIDIA GPUs.

Reply:
Just to add on the duplication and usage part. I think it makes more sense to centralize this under framework/utils.go as this means that if NVIDIA wants to change this name or google the COS name advertised then it won't have to go change that occurrence everywhere

Reply:
Can we just add DsFromManifest() in this file and keep other definitions local in test/e2e_node/device-plugin.go? It is ok to have some dup code for now. We can add a common pkg later if we see a lot duplicate functionalities.

Reply:
I'm sorry but I seem to fail seeing the benefits of having this code be duplicated in two separate files vs keeping it centralized in a place that seem easier to change.
Is there anything I am missing ?

Reply:
If you feel strong about avoiding duplicate code, can we put them into gpu_util.go? I think our main concern is these definitions are gpu specific and irrelevant to most people using e2e framework.

Reply:
I can put them in gpu_util.go I'll also add some comments :)
I think that should solve everyone's concerns